aml_interface/azure_logging.py

```diff
@@ -1,43 +1,112 @@
+        "logging": {
+            "loglevel_own": "INFO",
+            "own_packages": ["a", "b"],
+            "basic_config": {
+                "level": "WARNING",
+                "format": "%(asctime)s|||%(levelname)-8s|%(name)s|%(message)s",
+                "datefmt": "%Y-%m-%d %H:%M:%S"
+            },
+            "ai_instrumentation_key": "INSERT_KEY_HERE"
+        }
+    }
+
+    azureLoggingConfigurer = AzureLoggingConfigurer(config["logging"])
+    azureLoggingConfigurer.setup_oor_logging()
+    logger = logging.getLogger("a")
+    logger.info("Message with logger a")
     """
-    logging.basicConfig(**logging_cfg["basic_config"])
-    instrumentation_key = logging_cfg["ai_instrumentation_key"]
-    azure_log_handler = AzureLogHandler(connection_string=instrumentation_key)
-    azure_log_handler.setLevel(logging_cfg["loglevel_own"])
-
-    packages = logging_cfg["own_packages"] + ([pkg_name] if pkg_name else [])
-
-    for pkg in packages:
-        pkg_logger = logging.getLogger(pkg)
-        if pkg_logger.handlers:
-            print(f"Handler for {pkg} has been set already.")
-        else:
-            pkg_logger.addHandler(azure_log_handler)
-            print(f"pkg {pkg} has the following handlers: {pkg_logger.handlers}")
+
+    def __init__(self, logging_cfg: Dict[str, Any], pkg_name: str = None):
+        """
+        Initializes the AzureLoggingConfigurer instance with the provided logging configuration.
+
+        Parameters
+        ----------
+        logging_cfg : dict
+            Logging configuration settings.
+
+        pkg_name : str, optional
+            Additional package name to configure logging for.
+        """
+        self.logging_cfg = logging_cfg
+        self.packages = self.logging_cfg["own_packages"] + ([pkg_name] if pkg_name else [])
+
+        self.instrumentation_key = self.logging_cfg["ai_instrumentation_key"]
+        self.azure_log_handler = AzureLogHandler(connection_string=self.instrumentation_key)
+        self.azure_log_handler.setLevel(logging_cfg["loglevel_own"])
+
+    def _setup_logging(self, additional_handlers: List[logging.Handler] = None):
+        """
+        Sets up basic logging configurations for the specified packages.
+
+        This method sets up logging configurations for the specified packages, including the addition of
+        AzureLogHandler for sending logs to Azure Application Insights, and optional additional handlers.
+
+        Parameters
+        ----------
+        additional_handlers : list of logging.Handler, optional
+                              Additional logging handlers to be added to the logger.
+        """
+        additional_handlers = additional_handlers or []
+        for pkg in self.packages:
+            pkg_logger = logging.getLogger(pkg)
+            pkg_logger.setLevel(self.logging_cfg["loglevel_own"])
+
+            if len(pkg_logger.handlers) == len(additional_handlers) + 1:
+                print(f"Handlers for {pkg} have been set already.")
```

Comment:
Is it necessary to print this? Or we can just invert the check, add them if not there and otherwise do nothing?